server/enterprise/metrics/metrics.go

```diff
@@ -0,0 +1,1369 @@
+			if err != nil {
+				m.Platform.Log().Warn("ReplicaLagTime query returned error", mlog.Err(err))
+			}
+		}
+
+		return float64(m.Platform.Store.TotalReadDbConnections())
+	})
+	m.Registry.MustRegister(m.DbReadConnectionsGauge)
+
+	m.DbSearchConnectionsGauge = prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Namespace:   MetricsNamespace,
+		Subsystem:   MetricsSubsystemDB,
+		Name:        "search_replica_connections_total",
+		Help:        "The total number of connections to the search replica database.",
+		ConstLabels: additionalLabels,
+	}, func() float64 { return float64(m.Platform.Store.TotalSearchDbConnections()) })
+	m.Registry.MustRegister(m.DbSearchConnectionsGauge)
+
+	m.DbReplicaLagGaugeAbs = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace:   MetricsNamespace,
+			Subsystem:   MetricsSubsystemDB,
+			Name:        "replica_lag_abs",
+			Help:        "An abstract unit for measuring replica lag.",
+			ConstLabels: additionalLabels,
+		},
+		[]string{"node"},
+	)
+	m.Registry.MustRegister(m.DbReplicaLagGaugeAbs)
+
+	m.DbReplicaLagGaugeTime = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace:   MetricsNamespace,
+			Subsystem:   MetricsSubsystemDB,
+			Name:        "replica_lag_time",
+			Help:        "A time unit for measuring replica lag.",
+			ConstLabels: additionalLabels,
+		},
+		[]string{"node"},
+	)
+	m.Registry.MustRegister(m.DbReplicaLagGaugeTime)
+
+	// HTTP Subsystem
+
+	m.HTTPWebsocketsGauge = prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Namespace:   MetricsNamespace,
+		Subsystem:   MetricsSubsystemHTTP,
+		Name:        "websockets_total",
+		Help:        "The total number of websocket connections to this server.",
+		ConstLabels: additionalLabels,
+	}, func() float64 { return float64(m.Platform.TotalWebsocketConnections()) })
+	m.Registry.MustRegister(m.HTTPWebsocketsGauge)
+
+	m.HTTPRequestsCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace:   MetricsNamespace,
+		Subsystem:   MetricsSubsystemHTTP,
+		Name:        "requests_total",
+		Help:        "The total number of http API requests.",
+		ConstLabels: additionalLabels,
+	})
+	m.Registry.MustRegister(m.HTTPRequestsCounter)
+
+	m.HTTPErrorsCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace:   MetricsNamespace,
+		Subsystem:   MetricsSubsystemHTTP,
+		Name:        "errors_total",
+		Help:        "The total number of http API errors.",
+		ConstLabels: additionalLabels,
+	})
+	m.Registry.MustRegister(m.HTTPErrorsCounter)
+
+	// Cluster Subsystem
+
+	m.ClusterHealthGauge = prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Namespace:   MetricsNamespace,
+		Subsystem:   MetricsSubsystemCluster,
+		Name:        "cluster_health_score",
+		Help:        "A score that gives an idea of how well it is meeting the soft-real time requirements of the gossip protocol.",
+		ConstLabels: additionalLabels,
+	}, func() float64 {
+		if m.Platform.Cluster() == nil {
+			return 0
+		}
```

Comment:
One minor change here not to assume `m.Platform.Cluster()` since we can now run this in development environments without that enterprise feature.